test/glue/candlepin/owner_test.rb

```diff
@@ -22,6 +22,7 @@ def teardown
     end
 
     def test_update_candlepin_owner_service_level
+      @org.stubs(:owner_details).returns(:load_owner_details)
```

Comment:
I'm not sure why, but `@org.owner_details` returns nil. Running with recording on doesn't solve the issue. I've stubbed it here to return the lazy_accessor initializer, load_owner_details.

Reply:
It looks like this has something to do with the fact that that org hasn't been persisted to the database and perhaps the way LazyAccessor works.

When I changed CandlepinOwnerSupport.create_organization to create! the org, it works in the sense that it raises VCR::Errors::UnhandledHTTPRequestError

Part of me thinks we should not stub this, and change CandlepinOwnerSupport like I mentioned. What do you think @jlsherrill ?

Reply:
We technically shouldn't get any UnhandledHTTPRequestErrors because we didn't modify any external calls. I tend to think it's something to do with the LazyAccessor.

Reply:
This is the culprit: https://github.com/Katello/katello/blob/6710b15ce2c9a19a8b8e16615f422fb99644d822/app/lib/katello/lazy_accessor.rb#L137

We aren't using options[:unless] (dont think we should be) and so the second part of the ternary evaluates to true because the org in question is a new_record ... and then the next code executed is `instance_variable_get("@#{attr}")` which is returning our nil value.

I get why the code is written this way. Why fetch this value when we haven't even saved this record? So, I don't think that behavior should change especially because I don't know what kind of impact that'll have across the board. Still leaning toward adjusting the CandlepinOwnerSupport to create the org, record the cassettes again w/o your stub and see what happens then

Reply:
+1   i think that makes to sense to actually persist the object

Reply:
Hmm.. creating/destroying orgs is so painful because of all the fk relations. Stubbing `persisted?` achieves the same result. Would that work for y'all? If not, where do you propose creating/destroying the org? In this one test, or in the CandlepinOwnerSupport create/destroy methods?